ai21/clients/vertex/gcp_authorization.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import annotations
+
+from typing import Optional
+
+import google.auth
+from google.auth.credentials import Credentials
+from google.auth.transport.requests import Request
+
+
+class GCPAuthorization:
+    def __init__(self, credentials: Optional[Credentials] = None, project_id: Optional[str] = None):
+        self._credentials = credentials
+        self._project_id = project_id
+
+    def _set_auth(self):
+        credentials, loaded_project_id = google.auth.default(
```

Comment:
Maybe its worth wrapping this in a `try, except` to catch `DefaultCredentialsError` and raise an error with our own message and type

Reply:
We should probably do the same for AWS (not in this PR though)